controllers/servicebinding_controller.go

```diff
@@ -575,3 +575,28 @@ func (r *ServiceBindingReconciler) validateSecretNameIsAvailable(ctx context.Con
 	}
 	return nil
 }
+
+func (r *ServiceBindingReconciler) maintain(ctx context.Context, binding *v1alpha1.ServiceBinding, log logr.Logger) (ctrl.Result, error) {
+	shouldUpdateStatus := false
+	if binding.Generation != binding.Status.ObservedGeneration {
+		binding.SetObservedGeneration(binding.Generation)
+		shouldUpdateStatus = true
+	}
+
+	if _, err := r.getSecret(ctx, binding.Namespace, binding.Spec.SecretName); err != nil {
+		if apierrors.IsNotFound(err) && !isDelete(binding.ObjectMeta) {
+			log.Info(fmt.Sprintf("secret not found recovering binding %s", binding.Name))
+			binding.Status.BindingID = ""
```

Comment:
don't you prefer to just fetch the binding and store the secret in this case?

Reply:
this logic already written it'll happen next reconcile I think it cleaner that way